lib/services/role.go

```diff
@@ -291,6 +292,24 @@ func filterInvalidWindowsLogins(candidates []string) []string {
 	return output
 }
 
+func warnInvalidAzureIdentities(candidates []string) {
+	for _, candidate := range candidates {
+		if !MatchValidAzureIdentity(candidate) {
+			log.Warningf("Invalid format of Azure identity %q", candidate)
+		}
+	}
+}
+
+var azureIdentityPattern = regexp.MustCompile(`(?i)^/subscriptions/([a-fA-F0-9-]+)/resourceGroups/([0-9a-zA-Z-]+)/providers/Microsoft\.ManagedIdentity/userAssignedIdentities/([0-9a-zA-Z-]+)$`)
```

Comment:
Yes. The biggest reason: `ParseResourceID` is far too lenient for my taste. There are various oddities, but to give you an example:

`///subscriptions///1020304050607-cafe-8090-a0b0c0d0e0f0///resourceGroups/example-resource-group/providers/Microsoft.ManagedIdentity/userAssignedIdentities/teleport-azure` 

will be happily parsed the same as

`/subscriptions/1020304050607-cafe-8090-a0b0c0d0e0f0/resourceGroups/example-resource-group/providers/Microsoft.ManagedIdentity/userAssignedIdentities/teleport-azure`.

After parsing, there is no indication that one had extra slashes sprinkled in. If we were to accept both, then either we would have to compare them after parsing - which would complicate our RBAC code quite a bit - or we would have to accept that `deny(foo)` may not work as expected simply because someone thrown in an extra slash somewhere.

In general, it tightens the logic by requiring a stricter format, which ultimately leads to more secure code as some classes of errors are eliminated.

Still, I think it is worth documenting this and testing that our function is stricter indeed: https://github.com/gravitational/teleport/pull/18675/commits/84831de0badfb8ad8a85015e838e29f9598f66d3